### PR TITLE
Added eth_signTypedData

### LINKED ISSUE #209: Document `eth_signTypedData`
Document the `eth_signTypedData` RPC method on this page: https://docs.web3signer.consensys.net/reference/api/json-rpc.

The official EIP-712 spec might help: https://eips.ethereum.org/EIPS/eip-712#specification-of-the-eth_signtypeddata-json-rpc

See https://github.com/Consensys/web3signer/pull/893.

### PR DESCRIPTION
**Pull request by MadelineMurray**
Fixes #209 

 Note - I still have to test my example before this is merged.

### CODE REVIEW

File: docs/reference/api/json-rpc.md [hunk 1]
```diff
@@ -129,6 +129,80 @@ curl -X POST --data '{"jsonrpc":"2.0","method":"eth_signTransaction","params":[{
...
+### Returns
+
+`DATA` - Signature
+
+<!--tabs-->
+
+# curl HTTP request
+
+```bash
+curl -X POST --data '{"jsonrpc":"2.0","method":"eth_signTypedData","params":["0xCD2a3d9F938E13CD947Ec05AbC7FE734Df8DD826", {"types":{"EIP712Domain":[{"name":"name","type":"string"},{"name":"version","type":"string"},{"name":"chainId","type":"uint256"},{"name":"verifyingContract","type":"address"}],"Person":[{"name":"name","type":"string"},{"name":"wallet","type":"address"}],"Mail":[{"name":"from","type":"Person"},{"name":"to","type":"Person"},{"name":"contents","type":"string"}]},"primaryType":"Mail","domain":{"name":"Ether Mail","version":"1","chainId":1,"verifyingContract":"0xCcCCccccCCCCcCCCCCCcCcCccCcCCCcCcccccccC"},"message":{"from":{"name":"Cow","wallet":"0xCD2a3d9F938E13CD947Ec05AbC7FE734Df8DD826"},"to":{"name":"Bob","wallet":"0xbBbBBBBbbBBBbbbBbbBbbbbBBbBbbbbBbBbbBBbB"},"contents":"Hello, Bob!"}}],"id":1}'
```
Comment (MadelineMurray) [inline docs/reference/api/json-rpc.md:191]: @Gabriel-Trintinalia - I added this example before submitting for review (forgot to mention in the comments) -> will this work?  Or did you see this and think another example is required?

Comment (Gabriel-Trintinalia) [reply docs/reference/api/json-rpc.md:191]: tagging @gfukushima :)

Comment (MadelineMurray) [reply docs/reference/api/json-rpc.md:191]: @Gabriel-Trintinalia - sorry!!

Comment (gfukushima) [reply docs/reference/api/json-rpc.md:191]: I guess I missed the example then!

**Review by gfukushima — APPROVED** [review]

**Pull request merged**